crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -652,44 +898,91 @@ ArrayNode stringArray(Iterable<String> list) {
         return arrayNode;
     }
 
-    @SuppressWarnings("unchecked")
-    public static void main(String[] args) throws IOException, ClassNotFoundException {
-        boolean yaml = false;
-        Map<String, String> labels = new LinkedHashMap<>();
+    static class CommandOptions {
+        private boolean yaml = false;
+        private LinkedHashMap<String, String> labels = new LinkedHashMap<>();
+        VersionRange<KubeVersion> targetKubeVersions = null;
+        ApiVersion crdApiVersion = null;
+        List<ApiVersion> apiVersions = null;
+        ApiVersion storageVersion = null;
         Map<String, Class<? extends CustomResource>> classes = new HashMap<>();
-        for (int i = 0; i < args.length; i++) {
-            String arg = args[i];
-            if (arg.startsWith("--")) {
-                if (arg.equals("--yaml")) {
-                    yaml = true;
-                } else if (arg.equals("--label")) {
-                    i++;
-                    int index = args[i].indexOf(":");
-                    if (index == -1) {
-                        argParseErr("Invalid --label " + args[i]);
-                    }
-                    labels.put(args[i].substring(0, index), args[i].substring(index + 1));
 
+        @SuppressWarnings({"unchecked", "CyclomaticComplexity"})
+        public CommandOptions(String[] args) throws ClassNotFoundException {
+            for (int i = 0; i < args.length; i++) {
+                String arg = args[i];
+                if (arg.startsWith("--")) {
+                    if (arg.equals("--yaml")) {
+                        yaml = true;
+                    } else if (arg.equals("--label")) {
```

Comment:
Just curious, but wouldn't be better to use `switch` construction for the `else if`?